kernel/asm/io.c

```diff
@@ -5,50 +5,47 @@
  * See the file "LICENSE" in the source distribution for details *
  *****************************************************************/
 
-#ifndef _IO_C
-#define _IO_C
-
 #include <types.h>
 
-static inline uint8_t asm_in_b(uint16_t port){
```

Comment:
With optimizations turned off (which I've done to avoid unpredictable complications with debugging), GCC doesn't inline functions unless it's explicitly told to.  I made the IO functions inline because, while it's not an issue yet, the function call overhead may cause performance issues when reading or writing large amounts of data via IO ports, when that's necessary for devices that don't support DMA.

Reply:
I'll try making it inline again and see if it works.

Reply:
Oh, was inlining causing issues?

Reply:
Ya.  I'm trying to fix them  now.   Given that's inlining is a performance issue can we pull this as-is and I'll submit a ticket to fix it?

Reply:
Gotcha, that sounds like a plan then.

Reply:
Issue added.  Thanks for your flexibility.